buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1263,7 +1263,41 @@ public String toString(int index, int length, Charset charset) {
 
     @Override
     public int indexOf(int fromIndex, int toIndex, byte value) {
-        return ByteBufUtil.indexOf(this, fromIndex, toIndex, value);
+        if (fromIndex <= toIndex) {
+            return firstIndexOf(fromIndex, toIndex, value);
+        } else {
+            return lastIndexOf(fromIndex, toIndex, value);
+        }
+    }
+
+    private int firstIndexOf(int fromIndex, int toIndex, byte value) {
+        fromIndex = Math.max(fromIndex, 0);
+        if (fromIndex >= toIndex || capacity() == 0) {
```

Comment:
replace the `if` with `checkIndex(toIndex - 1)`

Reply:
hmm... This code is the same as in the existing method. So I don't think we should change behaviour:

https://github.com/netty/netty/blob/4.1/buffer/src/main/java/io/netty/buffer/ByteBufUtil.java#L459